Update fuzzer dictionary

pamarcos opened the pull request:
Update fuzzer dictionary as per commit 99282e526a8aeb175e9f3e69fc9385

- Use `LC_ALL=C sort` so that the result is consistent for both macOS and Linux. Turns out macOS's `sort` is of course the FreeBSD version which by default mixes strings starting with capital letters with lowercase ones, while Linux sorts first the uppercase ones:

macOS
```
Apple
apple
Banana
banana
```

Linux
```
Apple
Banana
apple
banana
```

With `LC_ALL=C` we'll get the same behavior independently of the OS and the locale.

- Create the `update_dict.sh` script that automates updating the fuzzer dictionary
- Remove the README since it's automated now

### Changelog category (leave one):
- Not for changelog (changelog entry is not required)


> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ] <!---ci_set_required--> Allow: All Required Checks
- [ ] <!---ci_include_stateless--> Allow: Stateless tests
- [ ] <!---ci_include_stateful--> Allow: Stateful tests
- [ ] <!---ci_include_integration--> Allow: Integration Tests
- [ ] <!---ci_include_performance--> Allow: Performance tests
- [ ] <!---ci_set_builds--> Allow: All Builds
- [ ] <!---batch_0_1--> Allow: batch 1, 2 for multi-batch jobs
- [ ] <!---batch_2_3--> Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ] <!---ci_exclude_style--> Exclude: Style check
- [ ] <!---ci_exclude_fast--> Exclude: Fast test
- [ ] <!---ci_exclude_asan--> Exclude: All with ASAN
- [ ] <!---ci_exclude_tsan|msan|ubsan|coverage--> Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ] <!---ci_exclude_aarch64|release|debug--> Exclude: All with aarch64, release, debug
---
- [ ] <!---do_not_test--> Do not test
- [ ] <!---woolen_wolfdog--> Woolen Wolfdog
- [ ] <!---upload_all--> Upload binaries for special builds
- [ ] <!---no_merge_commit--> Disable merge-commit
- [ ] <!---no_ci_cache--> Disable CI cache

pamarcos committed:
Update fuzzer dictionary as per commit 99282e526a8aeb175e9f3e69fc9385070d03798a

Also update README so that we have consistent behavior of sort
across macOS and Linux.

pamarcos committed:
Write a script to automatically update the dict

robot-clickhouse commented:
<!-- automatic status comment for PR #68284 from pamarcos/ClickHouse:update-fuzzer-dict -->
*This is an automated comment for commit 79e055783931bd544897ec23eb07b7c7c9b09a69 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/68284/79e055783931bd544897ec23eb07b7c7c9b09a69/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Builds</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/68184/35548d6a652c3e3bd6b018e00a2acd4e9c1ad056/builds/report.html">✅ success</a></td></tr>
<tr><td>ClickBench</td><td>Runs [ClickBench](https://github.com/ClickHouse/ClickBench/) with instant-attach table</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/99282e526a8aeb175e9f3e69fc9385070d03798a/clickbench__aarch64_.html">✅ success</a></td></tr>
<tr><td>Compatibility check</td><td>Checks that <b>clickhouse</b> binary runs on distributions with old libc versions. If it fails, ask a maintainer for help</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/99282e526a8aeb175e9f3e69fc9385070d03798a/compatibility_check__aarch64_.html">✅ success</a></td></tr>
<tr><td>Docker keeper image</td><td>The check to build and optionally push the mentioned image to docker hub</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/99282e526a8aeb175e9f3e69fc9385070d03798a/docker_keeper_image.html">✅ success</a></td></tr>
<tr><td>Docker server image</td><td>The check to build and optionally push the mentioned image to docker hub</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/99282e526a8aeb175e9f3e69fc9385070d03798a/docker_server_image.html">✅ success</a></td></tr>
<tr><td>Fast test</td><td>Normally this is the first check that is ran for a PR. It builds ClickHouse and runs most of <a href="https://clickhouse.com/docs/en/development/tests#functional-tests">stateless functional tests</a>, omitting some. If it fails, further checks are not started until it is fixed. Look at the report to see which tests fail, then reproduce the failure locally as described <a href="https://clickhouse.com/docs/en/development/tests#functional-test-locally">here</a></td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/99282e526a8aeb175e9f3e69fc9385070d03798a/fast_test.html">✅ success</a></td></tr>
<tr><td>Install packages</td><td>Checks that the built packages are installable in a clear environment</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/99282e526a8aeb175e9f3e69fc9385070d03798a/install_packages__aarch64_.html">✅ success</a></td></tr>
<tr><td>Stateful tests</td><td>Runs stateful functional tests for ClickHouse binaries built in various configurations -- release, debug, with sanitizers, etc</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/99282e526a8aeb175e9f3e69fc9385070d03798a/stateful_tests__aarch64_.html">✅ success</a></td></tr>
<tr><td>Stateless tests</td><td>Runs stateless functional tests for ClickHouse binaries built in various configurations -- release, debug, with sanitizers, etc</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/99282e526a8aeb175e9f3e69fc9385070d03798a/stateless_tests__aarch64_.html">✅ success</a></td></tr>
<tr><td>Stress test</td><td>Runs stateless functional tests concurrently from several clients to detect concurrency-related errors</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/99282e526a8aeb175e9f3e69fc9385070d03798a/stress_test__asan_.html">✅ success</a></td></tr>
<tr><td>Unit tests</td><td>Runs the unit tests for different release types</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/99282e526a8aeb175e9f3e69fc9385070d03798a/unit_tests__asan_.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

pamarcos committed:
Update dict files

Previous commit had the files from the script ran
from the root directory, which returned an error.

pamarcos commented:
Building the fuzzer failed, but luckily, the PR that fixes it was just merged: https://github.com/ClickHouse/ClickHouse/pull/61908
I'll merge master and re-trigger the build

pamarcos committed:
Merge branch 'master' into update-fuzzer-dict
